### PR TITLE
03 flow control is done

### PR DESCRIPTION
**Pull request by upl1nk-v01d**
all section's 'Flow Control' exercises are done.
note: on exercise GetTheCentury wrong commit message, should be committed as exercise 06.

### CODE REVIEW

**Review by Janchuks — APPROVED** [review]

File: csharp-basics/exercises/FlowOfControl/TenBillion/Program.cs
Comment (Latviets) [inline csharp-basics/exercises/FlowOfControl/TenBillion/Program.cs:1]: It kā nemanu kur tiktu veikta šī daļa: count the number of digits the number (...) has.

**Review by Latviets — APPROVED** [review]

File: csharp-basics/exercises/FlowOfControl/GetTheCentury/Program.cs
Comment (Janchuks) [inline csharp-basics/exercises/FlowOfControl/GetTheCentury/Program.cs:1]: You had some spare time to check for all the options. Super impressed !!

File: csharp-basics/exercises/FlowOfControl/PositiveNegativeNumber/Program.cs [hunk 1]
```diff
@@ -7,22 +7,17 @@ class Program
...
-            if (?)
-            {
+            var input = Console.ReadLine();
+        
+            if (double.Parse(input) > 0){
                 Console.WriteLine("Number is positive");
             } 
-            else if (?) 
-            {
+            else if (double.Parse(input) < 0){
```
Comment (Janchuks) [inline csharp-basics/exercises/FlowOfControl/PositiveNegativeNumber/Program.cs]: Why did you do the parsing to double?

File: csharp-basics/exercises/FlowOfControl/exc04/Program.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            }
+
+            err = false;
+
+            displayText("Please enter weekday number: ",30, 0);
+            prompt = Console.ReadLine()!; //added '!' to forgive null reference
+            if(rx.Matches(prompt).Count > 0){
+                err = true;
+                Console.Clear();
+                displayText("wtf? Please only a digit!");
```
Comment (Janchuks) [inline csharp-basics/exercises/FlowOfControl/exc04/Program.cs]: bahahahahahaha!! "wtf?" has me chuckling

File: csharp-basics/exercises/FlowOfControl/exc05/Program.cs
Comment (Janchuks) [inline csharp-basics/exercises/FlowOfControl/exc05/Program.cs:1]: This one was funny! Love itt!!

File: csharp-basics/exercises/FlowOfControl/exc07/Program.cs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        System.Threading.Thread.Sleep(delay);
+    }
+    static void clc(){
+        Console.Clear();
+    }
+    static void exit(int p){
+        System.Environment.Exit(p);
+    }
+
+    static void displayText(string text="\n", int tick=30, int el=1, int delay=0){
```
Comment (Janchuks) [inline csharp-basics/exercises/FlowOfControl/exc07/Program.cs]: Im lost.. What does the int el represent?

**Review by Janchuks — APPROVED** [review]
10/10. Some bits were super funny, which is totally adorable and made me smile. :))

**Review by EdgarsSalmins — CHANGES_REQUESTED** [review]
tas pats kas citos citos PR.. :) jāsataisa formatējums.. nosaukumi jāsalbo.. :)

File: csharp-basics/exercises/FlowOfControl/GetTheCentury/Program.cs [hunk 4]
```diff
@@ -1,12 +1,158 @@
...
+                int year = int.Parse(prompt);
+                
+                if(regexNotNumbers.Matches(prompt).Count > 0)
+                {
+                    error = true;
+
+                    Console.Clear();
+                    DisplayText("Only numbers are allowed here.");
+                } 
+
```
Comment (EdgarsSalmins) [inline csharp-basics/exercises/FlowOfControl/GetTheCentury/Program.cs:94]:
```suggestion
```

Comment (EdgarsSalmins) [reply csharp-basics/exercises/FlowOfControl/GetTheCentury/Program.cs:94]: šis uz visiem else if attiecas jo if un else if ir saistītas lietas. :)

File: csharp-basics/exercises/FlowOfControl/LargestNumber/Program.cs [hunk 5]
```diff
@@ -1,10 +1,25 @@
...
+            double largestNumber = 0;
+
+            foreach(double element in arrayNumbers)
+            {
+                if(element > largestNumber)
+                {
+                    largestNumber = element;
+                }
+            }
+            return largestNumber;
```
Comment (EdgarsSalmins) [inline csharp-basics/exercises/FlowOfControl/LargestNumber/Program.cs:20]:
```suggestion

            return largestNumber;
```

File: csharp-basics/exercises/FlowOfControl/exc04/Program.cs [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+                {
+                    quit = true;
+                }
+            }
+            
+            if(quit)
+            {
+                Clear();
+                DisplayText("Goodbye! :)");
+                Sleep(1000);
```
Comment (EdgarsSalmins) [inline csharp-basics/exercises/FlowOfControl/exc04/Program.cs:137]: kam Tev tas sleep ir vajadzīgs?

**Review by EdgarsSalmins — COMMENTED** [review]
man tas sleep sāk ļoti interesēt.. priekš kam? visos darbos redzu.

**Review by EdgarsSalmins — APPROVED** [review]

**Pull request merged**